client/network/src/service.rs

```diff
@@ -1148,6 +1242,8 @@ where
 	/// For each peer and protocol combination, an object that allows sending notifications to
 	/// that peer. Shared with the [`NetworkService`].
 	peers_notifications_sinks: Arc<Mutex<HashMap<(PeerId, ProtocolName), NotificationsSink>>>,
+	/// Peer reputation store handle.
+	peer_store_handle: PeerStoreHandle,
```

Comment:
> That requires some additional work with the NetworkServiceProvider contraption syncing uses but I'll deal with that in my PR.

Do you want me to leave it as is in this PR?

Logically `NetworkService::peer_store()` fits into `NetworkPeers` trait, but that one has a blanket implementation for `self: Arc<T>`, which makes it impossible to provide write access to `PeerStore`.

Reply:
I can implement it for `NetworkServiceProvider` unless you absolutely want to do it but `NetworkService` must have a getter for `PeerStoreHandle`. If there are issues with the blanket implementation, another way to do it would be to store `PeerStore` in `FullNetworkConfiguration` and giving the `PeerStoreHandle` to every protocol that might need it. That'd imply undoing some of the work of this PR which we don't need to now if you can implement a getter for `PeerStoreHandle`.

Reply:
I can leave it for you to implement in `NetworkServiceProvider`. It would be trivial to add it now to `NetworkService` impl, but we better put it into a trait that is easily distributable among protocols.

Reply:
LGTM